ultratrace2/model/project.py

```diff
@@ -1,23 +1,68 @@
+import logging
 import os
+import pickle
 
 from .trace import TraceList
 from .files.bundle import FileBundleList
 
+logger = logging.getLogger(__name__)
+
 
 class Project:
-    def __init__(self, path: str):
-        if not os.path.exists(path):
-            raise ValueError(f"cannot initialize project at {path}")
-        self.root_path = os.path.realpath(os.path.abspath(path))  # absolute path
-        self.traces = TraceList()
-        self.files = FileBundleList(self.root_path)
+    def __init__(self, traces: TraceList, files: FileBundleList):
+        """
+        Internal function: to construct a Project, either call ::get_by_path()
+        """
+        self.traces = traces
+        self.files = files
 
     def save(self):
         raise NotImplementedError()
 
     @classmethod
-    def load(cls):
-        raise NotImplementedError()
+    def load(cls, save_file) -> "Project":
+        try:
+            with open(save_file, "rb") as fp:
+                project = pickle.load(fp)
+                assert isinstance(project, Project)
+                return project
+        except Exception as e:
+            logger.error(e)
+            raise RuntimeError(str(e))
```

Comment:
Any reason to force everything to become a `RuntimeError`? Why not just
```suggestion
            raise
```

Reply:
I want to specifically catch one type of error in the caller.  I figured `RuntimeError` was the most generic `Exception` subclass.

Reply:
In that case you can still do `except Exception` at the caller without erasing the type of the originally thrown exception

Reply:
Done in 7450611